src/components/dialog-full-screen/dialog-full-screen.component.js

```diff
@@ -1,139 +1,97 @@
-    this.document.documentElement.style.overflow = this.originalOverflow;
-    return this.document.documentElement;
-  }
-
-  /**
-   * Returns HTML and text for the dialog title.
-   */
-  dialogTitle = () => {
-    let { title } = this.props;
-
-    if (typeof title === "string") {
-      title = (
+const DialogFullScreen = ({
+  open,
+  children,
+  title,
+  subtitle,
+  pagesStyling,
+  headerChildren,
+  showCloseIcon,
+  disableContentPadding,
+  disableEscKey,
+  onCancel,
+  ...rest
+}) => {
+  const dialogRef = useRef();
+  const headingRef = useRef();
+  const contentRef = useRef();
+
+  const dialogTitle = () => (
+    <FullScreenHeading
+      hasContent={title}
+      ref={headingRef}
+      showCloseIcon={showCloseIcon}
+      onCancel={onCancel}
+    >
+      {typeof title === "string" ? (
         <Heading
           title={title}
           titleId="carbon-dialog-title"
-          subheader={this.props.subtitle}
+          subheader={subtitle}
           subtitleId="carbon-dialog-subtitle"
           divider={false}
         />
-      );
-    }
-
-    return (
-      <FullScreenHeading
-        hasContent={title}
-        ref={this.headingRef}
-        showCloseIcon={this.props.showCloseIcon}
-        onCancel={this.props.onCancel}
-      >
-        {title}
-        {this.props.headerChildren}
-      </FullScreenHeading>
-    );
+      ) : (
+        title
+      )}
+      {headerChildren}
+    </FullScreenHeading>
+  );
+
+  const componentTags = {
+    "data-component": "dialog-full-screen",
+    "data-element": rest["data-element"],
+    "data-role": rest["data-role"],
   };
 
-  /**
-   * Returns the computed HTML for the dialog.
-   */
-  get modalHTML() {
-    return (
-      <FocusTrap wrapperRef={this._dialog}>
+  return (
+    <Modal
+      open={open}
+      onCancel={onCancel}
+      disableEscKey={disableEscKey}
+      {...componentTags}
+    >
+      <FocusTrap wrapperRef={dialogRef}>
         <StyledDialogFullScreen
-          ref={(d) => {
-            this._dialog = d;
-          }}
+          ref={dialogRef}
           data-element="dialog-full-screen"
-          pagesStyling={this.props.pagesStyling}
+          pagesStyling={pagesStyling}
         >
-          {this.dialogTitle()}
+          {dialogTitle()}
           <StyledContent
-            hasHeader={this.props.title !== undefined}
-            headingHeight={this.state.headingHeight}
+            hasHeader={title !== undefined}
             data-element="content"
-            ref={this.contentRef}
-            disableContentPadding={this.props.disableContentPadding}
+            ref={contentRef}
+            disableContentPadding={disableContentPadding}
           >
-            {this.props.children}
+            {children}
           </StyledContent>
         </StyledDialogFullScreen>
       </FocusTrap>
-    );
-  }
-}
+    </Modal>
+  );
+};
 
 DialogFullScreen.defaultProps = {
   open: false,
-  enableBackgroundUI: true,
   showCloseIcon: true,
```

Comment:
See comment about removing `showCloseIcon`, I don't think defaulting it to `true` does anything as they would still have to pass an `onCancel` callback